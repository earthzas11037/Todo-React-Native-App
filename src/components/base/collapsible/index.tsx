import React from 'react'
import { cssInterop } from 'nativewind'
import LibCollapsible, { CollapsibleProps as LibCollapsibleProps } from 'react-native-collapsible'

// Apply NativeWind integration
cssInterop(LibCollapsible, { className: 'style' })

interface CollapsibleProps extends LibCollapsibleProps {
  /**
   * Additional styles for the container
   */
  className?: string
  expanded: boolean
}

/**
 * Reusable Collapsible component with default props
 */
export const Collapsible: React.FC<CollapsibleProps> = ({ expanded, duration = 500, children, ...props }) => {
  return (
    <LibCollapsible collapsed={!expanded} duration={duration} {...props}>
      {children}
    </LibCollapsible>
  )
}
