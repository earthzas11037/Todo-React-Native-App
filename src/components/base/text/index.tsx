import { textTv } from '@/components/base/text/text.style'
import { forwardRef, useMemo } from 'react'
import type { TextProps, TextStyle } from 'react-native'
import { I18nManager, StyleSheet, Text as RNText, Platform } from 'react-native'
import { twMerge } from 'tailwind-merge'
import { type VariantProps } from 'tailwind-variants'

interface Props extends TextProps, VariantProps<typeof textTv> {
  className?: string
}

export const Text = forwardRef<RNText, Props>(({ className = '', style, children, variant, color, ...props }, ref) => {
  const textStyle = useMemo(() => twMerge(textTv({ variant, color }), className), [variant, color, className])

  const nStyle = useMemo(
    () =>
      StyleSheet.flatten([
        {
          writingDirection: I18nManager.isRTL ? 'rtl' : 'ltr'
        },
        style
      ]) as TextStyle,
    [style]
  )

  return (
    <RNText ref={ref} className={textStyle} style={nStyle} {...props}>
      {children}
    </RNText>
  )
})
