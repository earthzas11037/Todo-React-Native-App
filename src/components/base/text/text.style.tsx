import { Platform } from 'react-native'
import { tv } from 'tailwind-variants'

export const textTv = tv({
  base: 'font-notoSansThai text-base text-text-primary mt-[1px]', // Base styles
  variants: {
    variant: Platform.select({
      ios: {
        Header1: 'text-[19px] font-semibold',
        Header2: 'text-[19px] font-semibold',
        Title: 'text-[16px] font-semibold',
        Subtitle: 'text-[14px] font-semibold',
        Button: 'text-[14px] font-medium',
        Body: 'text-[14px] font-normal',
        CaptionSemiBold: 'text-[14px] font-semibold',
        CaptionRegular: 'text-[14px] font-normal',
        Support: 'text-[10px] font-semibold',
        Time: 'text-[10px] font-normal italic'
      },
      android: {
        Header1: `text-[21px] font-semibold`,
        Header2: 'text-[21px] font-semibold',
        Title: 'text-[18px] font-semibold',
        Subtitle: 'text-[16px] font-semibold',
        Button: 'text-[16px] font-medium',
        Body: 'text-[16px] font-normal',
        CaptionSemiBold: 'text-[16px] font-semibold',
        CaptionRegular: 'text-[16px] font-normal',
        Support: 'text-[12px] font-semibold',
        Time: 'text-[12px] font-normal italic'
      },
      default: {
        Header1: 'text-[19px] font-semibold',
        Header2: 'text-[19px] font-semibold',
        Title: 'text-[16px] font-semibold',
        Subtitle: 'text-[14px] font-semibold',
        Button: 'text-[14px] font-medium',
        Body: 'text-[14px] font-normal',
        CaptionSemiBold: 'text-[14px] font-semibold',
        CaptionRegular: 'text-[14px] font-normal',
        Support: 'text-[10px] font-semibold',
        Time: 'text-[10px] font-normal italic'
      }
    }),
    color: {
      primary: 'text-primary-500',
      secondary: 'text-secondary-500',
      success: 'text-success-500',
      error: 'text-error-500',
      warning: 'text-warning-500',
      info: 'text-info-500',
      textPrimary: 'text-text-primary',
      textSecondary: 'text-text-secondary',
      inherit: '' // Allows inherited color
    }
  },
  defaultVariants: {
    variant: 'Body', // Default variant
    color: 'textPrimary' // Default color
  }
})
