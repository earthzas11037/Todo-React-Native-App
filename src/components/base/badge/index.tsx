import { badgeTv } from '@/components/base/badge/badge.style'
import { Text } from '@/components/base/text'
import React from 'react'
import { View } from 'react-native'
import { tv, type VariantProps } from 'tailwind-variants'

interface BadgeProps extends VariantProps<typeof badgeTv> {
  content?: string | number
  invisible?: boolean
  children?: React.ReactNode
  className?: string
  badgeClassName?: string
  contentClassName?: string
}

export const Badge: React.FC<BadgeProps> = ({
  children,
  content,
  size = 'medium',
  color = 'primary',
  variant = 'standard',
  invisible = false,
  className,
  badgeClassName,
  contentClassName
}) => {
  const styles = React.useMemo(() => badgeTv({ size, color, variant, invisible }), [size, color, variant, invisible])
  const displayContent =
    typeof content === 'number' && variant !== 'dot' ? (content > 99 ? '99+' : content.toString()) : content

  return (
    <View className={`${styles.container({ className })}`}>
      {children}
      {!!content && (
        <View className={`${styles.badge({ className: badgeClassName })}`}>
          <Text className={`${styles.content({ className: contentClassName })}`}>
            {variant !== 'dot' ? displayContent : ''}
          </Text>
        </View>
      )}
    </View>
  )
}
