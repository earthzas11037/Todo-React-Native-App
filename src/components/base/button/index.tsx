import React, { JSX } from 'react'
import type { TouchableWithoutFeedbackProps } from 'react-native'
import { ActivityIndicator, TouchableWithoutFeedback, View } from 'react-native'
import type { VariantProps } from 'tailwind-variants'
import colors from '@/config/theme/colors'
import { buttonTv } from '@/components/base/button/button.style'
import { twMerge } from 'tailwind-merge'
import { LinearGradient } from '@/components/base/linear-gradient'
import { Text } from '@/components/base/text'

type ButtonVariants = VariantProps<typeof buttonTv>
interface Props extends ButtonVariants, Omit<TouchableWithoutFeedbackProps, 'disabled'> {
  label?: string
  loading?: boolean
  startIcon?: JSX.Element
  endIcon?: JSX.Element
  className?: string
  textClassName?: string
  iconClassName?: string
}

export const Button = React.forwardRef<View, Props>(
  (
    {
      label,
      loading = false,
      variant = 'contained',
      size = 'medium',
      color = 'primary',
      disabled = false,
      fullWidth = false,
      className = '',
      testID,
      textClassName = '',
      iconClassName = '',
      startIcon,
      endIcon,
      ...props
    },
    ref
  ) => {
    const [pressed, setPressed] = React.useState(false)

    const styles = React.useMemo(
      () => buttonTv({ variant, size, color, disabled, fullWidth }),
      [variant, size, color, disabled, fullWidth]
    )

    const isGradient = colors.gradient.primary && color === 'primary' && variant === 'contained' && disabled === false
    const gradientColor = colors.gradient.primary

    const handlePressIn = () => setPressed(true)
    const handlePressOut = () => setPressed(false)

    const dynamicStyle = pressed ? 'opacity-75' : ''

    if (isGradient) {
      return (
        <TouchableWithoutFeedback
          disabled={disabled || loading}
          {...props}
          testID={testID}
          onPressIn={handlePressIn}
          onPressOut={handlePressOut}
        >
          <LinearGradient
            ref={ref}
            {...(gradientColor as any)}
            className={twMerge(`${styles.container({ className })}`, dynamicStyle)}
          >
            {loading ? (
              <ActivityIndicator
                size='small'
                color='white'
                testID={testID ? `${testID}-activity-indicator` : undefined}
              />
            ) : (
              <>
                {startIcon &&
                  React.cloneElement(startIcon as any, {
                    className: `mr-[10px] ${styles.icon({ className: iconClassName })} ${
                      startIcon.props.className || ''
                    }`
                  })}
                <Text
                  testID={testID ? `${testID}-label` : undefined}
                  className={styles.label({ className: textClassName })}
                >
                  {label}
                </Text>
                {endIcon &&
                  React.cloneElement(endIcon as any, {
                    className: `ml-[10px] ${styles.icon({ className: iconClassName })} ${endIcon.props.className || ''}`
                  })}
              </>
            )}
          </LinearGradient>
        </TouchableWithoutFeedback>
      )
    }
    return (
      <TouchableWithoutFeedback
        disabled={disabled || loading}
        {...props}
        // ref={ref}
        testID={testID}
        onPressIn={handlePressIn}
        onPressOut={handlePressOut}
      >
        <View ref={ref} className={`${styles.container({ className })} ${dynamicStyle}`}>
          {loading ? (
            <ActivityIndicator
              size='small'
              className={styles.indicator()}
              testID={testID ? `${testID}-activity-indicator` : undefined}
            />
          ) : (
            <>
              {startIcon &&
                React.cloneElement(startIcon as any, {
                  className: `mr-[10px] ${styles.icon({ className: iconClassName })} ${startIcon.props.className || ''}`
                })}
              <Text
                testID={testID ? `${testID}-label` : undefined}
                className={styles.label({ className: textClassName })}
              >
                {label}
              </Text>
              {endIcon &&
                React.cloneElement(endIcon as any, {
                  className: `ml-[10px] ${styles.icon({ className: iconClassName })} ${endIcon.props.className || ''}`
                })}
            </>
          )}
        </View>
      </TouchableWithoutFeedback>
    )
  }
)
