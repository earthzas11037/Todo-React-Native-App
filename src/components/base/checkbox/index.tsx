import { MotiView } from 'moti'
import React, { useCallback, useEffect, useRef } from 'react'
import { Animated, I18nManager, Pressable, type PressableProps, View } from 'react-native'
import colors from '@/config/theme/colors'
import { Text } from '@/components/base/text'
import { VariantProps } from 'tailwind-variants'
import CustomIcon from '@/components/base/icon'
import { checkboxTv } from '@/components/base/checkbox/checkbox.style'

const SIZE = 18
const WIDTH = 50
const HEIGHT = 28
const THUMB_HEIGHT = 22
const THUMB_WIDTH = 22
const THUMB_OFFSET = 4

type CheckboxVariants = VariantProps<typeof checkboxTv>

export interface RootProps extends CheckboxVariants, Omit<PressableProps, 'onPress'> {
  onChange: (checked: boolean) => void
  checked?: boolean
  className?: string
  accessibilityLabel?: string
}

export type IconProps = {
  checked: boolean
  disabled?: boolean | null | undefined
  color?: 'primary' | 'secondary' | 'error' | 'info' | 'success' | 'warning'
}

export const Root = ({ checked = false, children, onChange, disabled, className = '', ...props }: RootProps) => {
  const handleChange = useCallback(() => {
    onChange(!checked)
  }, [onChange, checked])

  return (
    <Pressable
      onPress={handleChange}
      className={`flex-row items-center ${className} ${disabled ? 'opacity-50' : ''}`}
      accessibilityState={{ checked }}
      disabled={disabled}
      {...props}
    >
      {children}
    </Pressable>
  )
}

type LabelProps = {
  text: string
  className?: string
  testID?: string
}

const Label = ({ text, testID, className = '' }: LabelProps) => {
  return (
    <Text testID={testID} className={` ${className} pl-[6px]`}>
      {text}
    </Text>
  )
}

export const CheckboxIcon = ({ checked = false, color = 'primary', disabled }: IconProps) => {
  const _color = checked ? colors[color][500] : colors.text.disabled
  return (
    <MotiView
      style={{
        height: SIZE,
        width: SIZE,
        borderColor: _color,
        shadowColor: colors.primary[500],
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: checked ? 0.25 : 0,
        shadowRadius: 3.84,
        elevation: 5
      }}
      className='items-center justify-center rounded-[5px] border-2'
      from={{ backgroundColor: 'transparent', borderColor: '#CCCFD6' }}
      animate={{
        backgroundColor: checked ? _color : 'transparent',
        borderColor: _color
      }}
      transition={{
        backgroundColor: { type: 'timing', duration: 100 },
        borderColor: { type: 'timing', duration: 100 }
      }}
    >
      <MotiView
        from={{ opacity: 0 }}
        animate={{ opacity: checked ? 1 : 0 }}
        transition={{ opacity: { type: 'timing', duration: 100 } }}
      >
        <CustomIcon name='IconCheck' size={16} color='white' />
      </MotiView>
    </MotiView>
  )
}

const CheckboxRoot = ({ checked = false, children, ...props }: RootProps) => {
  return (
    <Root checked={checked} accessibilityRole='checkbox' {...props}>
      {children}
    </Root>
  )
}

const CheckboxBase = ({
  checked = false,
  testID,
  label,
  color = 'primary',
  ...props
}: RootProps & { label?: string }) => {
  return (
    <CheckboxRoot checked={checked} testID={testID} color={color} {...props}>
      <CheckboxIcon checked={checked} color={color} />
      {label ? <Label text={label} testID={testID ? `${testID}-label` : undefined} className='pr-2' /> : null}
    </CheckboxRoot>
  )
}

export const Checkbox = Object.assign(CheckboxBase, {
  Icon: CheckboxIcon,
  Root: CheckboxRoot,
  Label
})

export const RadioIcon = ({ checked = false, color = 'primary', disabled }: IconProps) => {
  const borderColor = checked ? colors[color][500] : disabled ? colors.gray.o.light : colors.text.disabled
  const innerColor = !disabled ? 'transparent' : checked ? colors[color][500] : colors.gray.o.light
  return (
    <View
      style={[
        {
          height: SIZE,
          width: SIZE,
          borderWidth: checked || (!checked && disabled) ? 5 : 2,
          justifyContent: 'center',
          alignItems: 'center',
          borderRadius: SIZE / 2,
          backgroundColor: 'transparent',
          shadowOffset: { width: 0, height: 2 },
          shadowRadius: 3.84,
          elevation: 5,
          borderColor: borderColor,
          shadowColor: disabled ? 'transparent' : colors[color][500],
          shadowOpacity: checked ? 0.25 : 0
        }
      ]}
    >
      {checked || disabled ? (
        <View
          style={[
            {
              height: 8,
              width: 8,
              borderRadius: 4
            },
            { backgroundColor: innerColor }
          ]}
        />
      ) : null}
    </View>
  )
}

const RadioRoot = ({ checked = false, children, ...props }: RootProps) => {
  return (
    <Root checked={checked} accessibilityRole='radio' {...props}>
      {children}
    </Root>
  )
}

const RadioBase = ({
  checked = false,
  testID,
  label,
  color = 'primary',
  disabled,
  ...props
}: RootProps & { label?: string }) => {
  return (
    <RadioRoot checked={checked} testID={testID} {...props}>
      <RadioIcon checked={checked} color={color} disabled={disabled} />
      {label ? <Label text={label} testID={testID ? `${testID}-label` : undefined} /> : null}
    </RadioRoot>
  )
}

export const Radio = Object.assign(RadioBase, {
  Icon: RadioIcon,
  Root: RadioRoot,
  Label
})

export const SwitchIcon = ({ checked = false }: IconProps) => {
  const translateX = useRef(new Animated.Value(checked ? THUMB_OFFSET : WIDTH - THUMB_WIDTH - THUMB_OFFSET)).current
  const backgroundColor = checked ? colors.primary[500] : colors.text.disabled

  useEffect(() => {
    Animated.timing(translateX, {
      toValue: checked ? THUMB_OFFSET : WIDTH - THUMB_WIDTH - THUMB_OFFSET,
      duration: 150,
      useNativeDriver: true
    }).start()
  }, [checked])

  return (
    <View className='w-[50px] justify-center'>
      <View className='overflow-hidden rounded-full'>
        <View
          style={{
            width: WIDTH,
            height: HEIGHT,
            backgroundColor
          }}
        />
      </View>
      <Animated.View
        style={{
          height: THUMB_HEIGHT,
          width: THUMB_WIDTH,
          position: 'absolute',
          backgroundColor: 'white',
          borderRadius: 13,
          right: 0,
          transform: [
            {
              translateX: I18nManager.isRTL ? translateX : Animated.multiply(translateX, -1)
            }
          ]
        }}
      />
    </View>
  )
}

const SwitchRoot = ({ checked = false, children, ...props }: RootProps) => {
  return (
    <Root checked={checked} accessibilityRole='switch' {...props}>
      {children}
    </Root>
  )
}

const SwitchBase = ({ checked = false, testID, label, ...props }: RootProps & { label?: string }) => {
  return (
    <SwitchRoot checked={checked} testID={testID} {...props}>
      <SwitchIcon checked={checked} />
      {label ? <Label text={label} testID={testID ? `${testID}-label` : undefined} /> : null}
    </SwitchRoot>
  )
}

export const Switch = Object.assign(SwitchBase, {
  Icon: SwitchIcon,
  Root: SwitchRoot,
  Label
})
