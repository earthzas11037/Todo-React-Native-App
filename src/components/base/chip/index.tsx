import React from 'react'
import { TouchableOpacity, Pressable } from 'react-native'
import { type VariantProps } from 'tailwind-variants'
import CustomIcon from '@/components/base/icon'
import { chipTv } from '@/components/base/chip/chip.style'
import { Text } from '@/components/base/text'

interface ChipProps extends VariantProps<typeof chipTv> {
  label: string
  icon?: JSX.Element
  deleteIcon?: JSX.Element
  onDelete?: () => void
  onPress?: () => void
  disabled?: boolean
  className?: string
  labelClassName?: string
  iconClassName?: string
  deleteIconClassName?: string
}

export const Chip: React.FC<ChipProps> = ({
  label,
  icon,
  deleteIcon,
  color = 'default',
  size = 'medium',
  disabled,
  variant = 'filled',
  fullWidth,
  onPress,
  onDelete,
  className,
  labelClassName,
  iconClassName,
  deleteIconClassName
}) => {
  const styles = React.useMemo(
    () => chipTv({ color, size, disabled, variant, fullWidth }),
    [color, size, disabled, variant, fullWidth]
  )

  return (
    <Pressable className={`${styles.container({ className })}`} onPress={onPress} disabled={disabled}>
      {icon &&
        React.cloneElement(icon as any, {
          className: `${styles.icon({ className: iconClassName })} ${icon.props.className || ''}`
        })}
      <Text className={`${styles.label({ className: labelClassName })}`}>{label}</Text>
      {onDelete && (
        <TouchableOpacity onPress={onDelete}>
          {deleteIcon ? (
            React.cloneElement(deleteIcon as any, {
              className: `${styles.deleteIcon({ className: deleteIconClassName })} ${deleteIcon.props.className || ''}`
            })
          ) : (
            <CustomIcon
              name='IconCircleX'
              size={size == 'small' ? 16 : size == 'medium' ? 20 : 24}
              className={`${styles.deleteIcon({ className: deleteIconClassName })}`}
            />
          )}
        </TouchableOpacity>
      )}
    </Pressable>
  )
}
