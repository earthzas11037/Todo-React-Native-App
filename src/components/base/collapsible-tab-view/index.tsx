import React from 'react'
import {
  Tabs,
  MaterialTabBar,
  CollapsibleProps,
  MaterialTabBarProps,
  MaterialTabItemProps,
  MaterialTabItem,
  TabItemProps
} from 'react-native-collapsible-tab-view'
import { Dimensions } from 'react-native'
import colors from '@/config/theme/colors'
import { Text } from '@/components/base/text'
import { twMerge } from 'tailwind-merge'

// Custom TabBar Component
const CustomTabBar: React.FC<MaterialTabBarProps<any>> = props => {
  const { width } = Dimensions.get('window')
  return (
    <MaterialTabBar
      {...props}
      activeColor={colors.primary[500]}
      inactiveColor={colors.text.primary}
      style={{
        backgroundColor: 'transparent'
      }}
      contentContainerStyle={{}}
      labelStyle={{}}
      tabStyle={{}}
      TabItemComponent={({ focusable, ...rest }: MaterialTabItemProps<any>) => {
        return (
          <MaterialTabItem
            {...rest}
            activeColor={colors.primary[500]}
            label={(itemProps: TabItemProps<any>) => {
              const isFocused = props.focusedTab.value == itemProps.name
              return (
                <Text
                  className={twMerge(`text-center px-5 py-2`)}
                  style={{ width: width / props.tabNames.length, color: props.activeColor }}
                >
                  {itemProps.name as String}
                </Text>
              )
            }}
          />
        )
      }}
      indicatorStyle={{
        backgroundColor: colors.primary[500],
        shadowOpacity: 0
      }}
      scrollEnabled
    />
  )
}

// Custom Tabs.Container with default TabBar customization
const CustomTabsContainer: React.FC<CollapsibleProps> = props => {
  return (
    <Tabs.Container
      {...props}
      renderTabBar={tabBarProps => <CustomTabBar {...tabBarProps} />}
      headerContainerStyle={{ shadowOpacity: 0, borderColor: colors.divider, borderBottomWidth: 1 }}
    />
  )
}

// Re-export Tabs components
const CustomTabs = {
  Container: CustomTabsContainer,
  Tab: Tabs.Tab,
  ScrollView: Tabs.ScrollView,
  FlatList: Tabs.FlatList,
  SectionList: Tabs.SectionList,
  Lazy: Tabs.Lazy,
  FlashList: Tabs.FlashList,
  MasonryFlashList: Tabs.MasonryFlashList
}

export { CustomTabs }
export * from 'react-native-collapsible-tab-view' // Export all props and types
