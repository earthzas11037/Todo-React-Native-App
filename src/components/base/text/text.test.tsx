import React from 'react'
import { render, screen } from '@testing-library/react-native'
import { I18nManager, Text as RNText } from 'react-native'
import { Text } from '@/components/base/text' // ⚠️ แก้ path ให้ตรงโปรเจกต์คุณ

describe('Text (Presentational Component)', () => {
  afterEach(() => {
    jest.clearAllMocks()
    I18nManager.isRTL = false // รีเซ็ตค่า RTL ทุกเคส
  })

  /* -------------------------------------------------------------------------- */
  /* การแสดงผลพื้นฐาน                                                         */
  /* -------------------------------------------------------------------------- */
  it('should render children correctly', () => {
    render(<Text>สวัสดี</Text>)
    expect(screen.getByText('สวัสดี')).toBeOnTheScreen()
  })

  /* -------------------------------------------------------------------------- */
  /* ทดสอบค่า defaultVariants                                                 */
  /* -------------------------------------------------------------------------- */
  it('should apply default variant (Body) & color (textPrimary)', () => {
    render(<Text>Default</Text>)
    const node = screen.UNSAFE_getByType(RNText)

    expect(node.props.className).toContain('text-[14px]') // Body
    expect(node.props.className).toContain('text-text-primary') // textPrimary
  })

  /* -------------------------------------------------------------------------- */
  /* ทดสอบการ Override variant & color                                        */
  /* -------------------------------------------------------------------------- */
  it('should apply custom variant Header1 and secondary color', () => {
    render(
      <Text variant='Header1' color='secondary'>
        Header
      </Text>
    )
    const node = screen.UNSAFE_getByType(RNText)

    expect(node.props.className).toContain('text-[19px]')
    expect(node.props.className).toContain('text-secondary-500')
  })

  /* -------------------------------------------------------------------------- */
  /* ทดสอบการรวม className ภายนอก                                           */
  /* -------------------------------------------------------------------------- */
  it('should merge extra className properly', () => {
    render(<Text className='underline'>Underline</Text>)
    expect(screen.UNSAFE_getByType(RNText).props.className).toContain('underline')
  })

  /* -------------------------------------------------------------------------- */
  /* ทดสอบ ref forwarding                                                     */
  /* -------------------------------------------------------------------------- */
  it('should forward ref to underlying RNText', () => {
    const ref = React.createRef<RNText>()
    render(<Text ref={ref}>Ref test</Text>)
    expect(ref.current).toBeTruthy()
    expect(ref.current?.props.children).toBe('Ref test')
  })

  /* -------------------------------------------------------------------------- */
  /* Snapshot Testing (Variants & States)                                     */
  /* -------------------------------------------------------------------------- */
  describe('Snapshots for variants / states', () => {
    it('renders Header1 / primary variant correctly', () => {
      const { toJSON } = render(
        <Text variant='Header1' color='primary'>
          Header
        </Text>
      )
      expect(toJSON()).toMatchSnapshot()
    })

    it('renders Body / error color correctly', () => {
      const { toJSON } = render(
        <Text variant='Body' color='error'>
          Error text
        </Text>
      )
      expect(toJSON()).toMatchSnapshot()
    })

    it('renders CaptionRegular with custom className', () => {
      const { toJSON } = render(
        <Text variant='CaptionRegular' className='italic'>
          Caption
        </Text>
      )
      expect(toJSON()).toMatchSnapshot()
    })
  })
})
