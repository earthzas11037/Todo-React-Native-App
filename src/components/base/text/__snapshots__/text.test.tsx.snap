// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text (Presentational Component) Snapshots for variants / states renders Body / error color correctly 1`] = `
<Text
  className="font-notoSansThai mt-[1px] text-[14px] font-normal text-error-500"
  style={
    {
      "writingDirection": "ltr",
    }
  }
>
  Error text
</Text>
`;

exports[`Text (Presentational Component) Snapshots for variants / states renders CaptionRegular with custom className 1`] = `
<Text
  className="font-notoSansThai mt-[1px] text-[14px] font-normal text-text-primary italic"
  style={
    {
      "writingDirection": "ltr",
    }
  }
>
  Caption
</Text>
`;

exports[`Text (Presentational Component) Snapshots for variants / states renders Header1 / primary variant correctly 1`] = `
<Text
  className="font-notoSansThai mt-[1px] text-[19px] font-semibold text-primary-500"
  style={
    {
      "writingDirection": "ltr",
    }
  }
>
  Header
</Text>
`;
